src/main/java/game/ObjectManager.java

```diff
@@ -246,6 +262,54 @@ public void updateObjects(PlayerState ps) {
                 ce.setVelocity(newVel);
             }
         }
+        for (Element boardObject : boardObjects) {
+            if (boardObject instanceof PushableElement){
+                pushables.add(boardObject);
+            }
+        }
+
+        // Set PushableElement's new position when countered Player
+        loop : {for (Element boardObject : boardObjects) {
+            if (boardObject instanceof PushableElement) {
+
+                // When player meets the ball.
+                if (Point2D.equals(boardObject.getPos(), playerPos)) {
+                    pushables.remove(boardObject);
+                    System.out.println("player: " + playerPos.getX() + ", " + playerPos.getY());
+
+                    System.out.println("object: " + boardObject.getPos().getX() + "," + boardObject.getPos().getY());
+
+
```

Comment:
Also, just put this into some function called processPushables or something, this will make it more readable.